streamer/parser/a11y-metadata-parsing.md

```diff
@@ -0,0 +1,271 @@
+# Parsing Accessibility Metadata
+
+The goal of this document is to provide directions that each implementation of Readium can follow when parsing schema.org Accessibility Metadata in EPUB publications.
+
+When parsing a publication in the streamer we always use the most complex form for each metadata to harmonize our output.
+
+Related Repository: [Readium Web Publication Manifest](https://github.com/readium/webpub-manifest)
+
+References: 
+
+- http://kb.daisy.org/publishing/docs/metadata/schema-org.html
+- http://kb.daisy.org/publishing/docs/metadata/evaluation.html
+- https://www.w3.org/wiki/WebSchemas/Accessibility
+
+## AccessMode
+
+`accessMode` is a key whose value is an array of strings.
+
+It is expected in a conformant EPUB publication ([EPUB Accessibility 1.0](https://www.w3.org/Submission/epub-a11y/#sec-disc-package)).
+
+At the time of writing, known values are:
+
+- `auditory`
+- `tactile`
+- `textual`
+- `visual`
+- `chartOnVisual`
+- `chemOnVisual`
+- `colorDependent`
+- `diagramOnVisual`
+- `mathOnVisual`
+- `musicOnVisual`
+- `textOnVisual`
+
+### EPUB 2.x
+
+The array is created from the `meta` elements whose `name` attribute has the value `schema:accessMode`. The value of their `content` attribute is pushed to the array.
+
+### EPUB 3.X
+
+The array is created from the `meta` elements whose `property` attribute has the value `schema:accessMode`. Their value is pushed to the array.
+
+## AccessModeSufficient
+
+`accessModeSufficient` is a key whose value is an array of arrays and strings.
+
+The known values are the same as `accessMode`.
+
+### EPUB 2.x
+
+The array is created from the `meta` elements whose `name` attribute has the value `schema:accessModeSufficient`. 
+
+When the value of their `content` attribute is a single `accessMode`, it is pushed to the array. When it is a multiple, another array is created then pushed to the parent array.
+
+### EPUB 3.X
+
+The array is created from the `meta` elements whose `property` attribute has the value `schema:accessModeSufficient`. 
+
+When their value is a single `accessMode`, it is pushed to the array. When it is a multiple, another array is created then pushed to the parent array.
+
+## AccessibilitySummary
+
+`accessibilitySummary` is a key whose value is an array of localized strings (JSON-LD language maps).
```

Comment:
There are other instances in the R2 models where "localized strings" can in fact be a single string, not necessarily a language-keyed object map. For example here, if there is only a single `schema:accessibilitySummary` and no `xml:lang` override is specified, then `accessibilitySummary` can simply be a string. Otherwise, `accessibilitySummary` must be an object keyed by locale, include `und` as fallback. Here is a hypothetical example (assuming `en-UK` is not explicit via `xml:lang`, thus the `und` key):
```
"accessibilitySummary": {
    "fr": "l'aubergine",
    "en-US": "the eggplant",
     "und": "the aubergine",
}
```

Reply:
JSON Schema example: https://github.com/readium/webpub-manifest/blob/d7c407953c3f7bef9665ceb482f36860ab5cf705/schema/metadata.schema.json#L15-L32

Reply:
Note that the EPUB parsing doc may need updating to reflect the "single string" possibility (which is in fact a common default, as titles are rarely provided in multiple languages)

https://github.com/readium/architecture/blob/master/streamer/parser/metadata.md#title

Reply:
PS: maybe *I* am out of date, because of recent changes related to SortAs etc. Please let me know :)

Reply:
That’s a good question, I’ll have to check since I can recall #124 and #125 ignited a discussion on that point but then I missed two weeks of comments, developments, and engineering calls. 

@mickael-menu could you perhaps remind me what the resolution was? I can vaguely recall the most complex form should be used in the streamer for interoperability reasons but not much more.

Reply:
the conclusion of the discussion on sortAs was indeed that it becomes a Localized string property, and that "und" indicates an undefined language (which should not be met in practice). The in-memory structure may vary between implementations, but generically it should be a "map" (plain strings are therefore mapped using the generic "language" property)

Reply:
Well, my Bible is the JSON Schema, so ... ;o)

Reply:
The point here is: is the "single string" value still allowed for potentially-multilingual metadata? (if so, this PR must be updated to reflect that, if not, the JSON Schema and other EPUB parsing doc must be updated, I think)

Reply:
yes a single string is allowed in the (serialized) RWPM, and the json schema highlights it (I was describing the normalized in-memory model in my previous comment).

Reply:
The second para at the top of the document seems to indicate that we're describing here the (normalized) in-memory model, not the serialized json. Isn't it the case?

Reply:
Okay, so I suppose the "parsing" documentation ; strictly-speaking ; relates to populating the internal data model, more than to the specificities of the serialized JSON format (normalized / canonical). However it can be quite confusing for newcomers to see that the common "single string" instance is not represented / mentioned in the documentation?

Reply:
Maybe we could add in this second para that for serialisation one can refer to the JSON schemas? (and same for the EPUB metadata PR?)

Reply:
well, the best we can do is being very clear at the top of the document that its focus is the mapping of EPUB 2/3 to the internal Readium Publication model, and that marshalled json RWPM is out of scope in this specification.

Reply:
Note that internally, the `r2-shared-js` TypeScript model for "potentially multilingual strings" is a union type of `string`, and the "language map" object (with locale keys). There is no normalization to "always an object, never a simple string".

Reply:
Thanks for the clarification, @llemeurf. I’ll reword the 2nd para and update this section.

(I guess we should also request a change in PR #125 then).

Reply:
Just committed some changes. Please let me know if that makes sense.

Reply:
I think we just mentioned that the output format from the in-memory model should be the most complex form. So if it's parsed from a single string, then when serialized again to JSON, we get `{"und": "my string"}`. This serves no purpose besides canonicalization.